packages/subscribable/src/__tests__/async-iterable-test.ts

```diff
@@ -0,0 +1,220 @@
+            const iterable = createAsyncIterableFromDataPublisher({
+                abortSignal: abortController.signal,
+                dataChannelName: 'data',
+                dataPublisher: mockDataPublisher,
+                errorChannelName: 'error',
+            });
+            iteratorA = iterable[Symbol.asyncIterator]();
+            iteratorB = iterable[Symbol.asyncIterator]();
+            nextDataPromiseA = iteratorA.next();
+            nextDataPromiseB = iteratorB.next();
+        });
+        it('throws from all iterators when an error is published', async () => {
+            expect.assertions(2);
+            publish('error', new Error('o no'));
+            await expect(nextDataPromiseA).rejects.toThrow(new Error('o no'));
+            await expect(nextDataPromiseB).rejects.toThrow(new Error('o no'));
+        });
+        it('vends a message to all iterators who have already polled for a result', async () => {
+            expect.assertions(2);
+            publish('data', 'hi');
+            await expect(nextDataPromiseA).resolves.toStrictEqual({
+                done: false,
+                value: 'hi',
+            });
+            await expect(nextDataPromiseB).resolves.toStrictEqual({
+                done: false,
+                value: 'hi',
+            });
+        });
+        it('queues messages for all iterators', async () => {
+            expect.assertions(4);
+            publish('data', 'consumed message');
+            publish('data', 'queued message 1');
+            publish('data', 'queued message 2');
+            await expect(iteratorA.next()).resolves.toStrictEqual({
+                done: false,
+                value: 'queued message 1',
+            });
+            await expect(iteratorB.next()).resolves.toStrictEqual({
+                done: false,
+                value: 'queued message 1',
+            });
+            await expect(iteratorA.next()).resolves.toStrictEqual({
+                done: false,
+                value: 'queued message 2',
+            });
+            await expect(iteratorB.next()).resolves.toStrictEqual({
+                done: false,
+                value: 'queued message 2',
+            });
+        });
+        it('flushes the queue before vending errors', async () => {
+            expect.assertions(4);
+            publish('data', 'consumed message');
+            publish('data', 'queued message 1');
+            publish('error', new Error('o no'));
+            await expect(iteratorA.next()).resolves.toStrictEqual({
+                done: false,
+                value: 'queued message 1',
+            });
+            await expect(iteratorB.next()).resolves.toStrictEqual({
+                done: false,
+                value: 'queued message 1',
+            });
+            await expect(iteratorA.next()).rejects.toThrow(new Error('o no'));
+            await expect(iteratorB.next()).rejects.toThrow(new Error('o no'));
+        });
```

Comment:
One key difference between this implementation and the last is that the last one would nuke the iterators immediately upon receiving an error. This might have caused code that was behind processing notifications to miss notifications that had already arrived on the client.

This implementation continues to vend any messages in the queue before vending the error. Errors are now a queue member like anything else.